### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by DankevichMisha**

### CODE REVIEW

File: jv-oop-advanced/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# jv-oop-advanced
```
Comment (bhdnchui) [inline jv-oop-advanced/README.md:1]: Redundant file.

File: jv-oop-advanced/checklist.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
+## Common mistakes (jv-oop-advanced)
```
Comment (bhdnchui) [inline jv-oop-advanced/checklist.md:1]: Redundant file.

File: jv-oop-advanced/pom.xml [hunk 3]
```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bhdnchui) [inline jv-oop-advanced/pom.xml:1]: Redundant file.

File: jv-oop-advanced/src/main/java/core/basesyntax/Circle.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private final int radius;
+
+    public Circle(Form randomForm, Color randomColor, int radius) {
+        super(randomForm,randomColor);
```
Comment (bhdnchui) [inline jv-oop-advanced/src/main/java/core/basesyntax/Circle.java:7]:
```suggestion
        super(randomForm, randomColor);
```

File: jv-oop-advanced/src/main/java/core/basesyntax/Drawable.java [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface Drawable {
+    Figure draw();
```
Comment (bhdnchui) [inline jv-oop-advanced/src/main/java/core/basesyntax/Drawable.java:4]:
```suggestion
    void draw();
```

File: jv-oop-advanced/src/main/java/core/basesyntax/Figure.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        return forms;
+    }
+
+    public void setForms(Form forms) {
+        this.forms = forms;
+    }
+
+    @Override
+    public Figure draw() {
+        System.out.println("не то что надо выводит! не пойму почему с figure использует");
```
Comment (bhdnchui) [inline jv-oop-advanced/src/main/java/core/basesyntax/Figure.java:30]: ?

File: jv-oop-advanced/src/main/java/core/basesyntax/FigureSupplier.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+    public static Form getRandomForm() {
+        int index = random.nextInt(forms.length);
+        return forms[index];
+    }
+
+    public static Figure getRandomFigure() {
+        side = random.nextInt(100);
+        firstleg = random.nextInt(100);
+        secondleg = random.nextInt(100);
```
Comment (bhdnchui) [inline jv-oop-advanced/src/main/java/core/basesyntax/FigureSupplier.java:21]: Make a constant.

File: jv-oop-advanced/src/main/java/core/basesyntax/FigureSupplier.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        Form randomForm = getRandomForm();
+        if (randomForm.name().equals("Circle")) {
```
Comment (bhdnchui) [inline jv-oop-advanced/src/main/java/core/basesyntax/FigureSupplier.java:23]:
```suggestion
        if (randomForm.equals(Form.CIRCLE)) {
```

File: jv-oop-advanced/src/main/java/core/basesyntax/Form.java [hunk 9]
```diff
@@ -0,0 +1,8 @@
+package core.basesyntax;
+
+public enum Form {
+    Circle,
+    Rectangle,
+    RightTriangle,
+    Square
```
Comment (bhdnchui) [inline jv-oop-advanced/src/main/java/core/basesyntax/Form.java:7]: Uppercase.

File: jv-oop-advanced/src/main/java/core/basesyntax/IsoscelesTrapezoid.java [hunk 10]
```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+
```
Comment (bhdnchui) [inline jv-oop-advanced/src/main/java/core/basesyntax/IsoscelesTrapezoid.java:4]:
```suggestion
```

File: jv-oop-advanced/src/main/java/core/basesyntax/Main.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[6];
```
Comment (bhdnchui) [inline jv-oop-advanced/src/main/java/core/basesyntax/Main.java:5]: Make constant.

File: jv-oop-advanced/src/main/java/core/basesyntax/Main.java [hunk 12]
```diff
@@ -0,0 +1,15 @@
...
+        for (int i = 0; i < 6; i++) { // констант
+            if (i < 3) {
+                figures[i] = FigureSupplier.getRandomFigure().draw();
+            }
+            if (i >= 3) {
+                figures[i] = FigureSupplier.getDefaultFigure().draw();
```
Comment (bhdnchui) [inline jv-oop-advanced/src/main/java/core/basesyntax/Main.java:11]: Make array and then draw figures.

File: jv-oop-advanced/src/test/java/core/test/StructureTest.java [hunk 13]
```diff
@@ -0,0 +1,155 @@
+package core.test;
```
Comment (bhdnchui) [inline jv-oop-advanced/src/test/java/core/test/StructureTest.java:1]: Redundant file.

File: jv-oop-advanced/target/checkstyle-cachefile [hunk 14]
```diff
@@ -0,0 +1,14 @@
+#Sat Oct 07 14:08:04 EEST 2023
```
Comment (bhdnchui) [inline jv-oop-advanced/target/checkstyle-cachefile:1]: Redundant file.

File: jv-oop-advanced/target/checkstyle-checker.xml [hunk 15]
```diff
@@ -0,0 +1,250 @@
+<?xml version="1.0"?>
```
Comment (bhdnchui) [inline jv-oop-advanced/target/checkstyle-checker.xml:1]: Redundant file.

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Good job, but fix comments.
Also, think about PR creation flow.

File: jv-oop-advanced/src/main/java/core/basesyntax/Circle.java (hunk 4 above)
Comment (DankevichMisha) [reply jv-oop-advanced/src/main/java/core/basesyntax/Circle.java:7]: I need to do one space? he is already there .. maybe mistake on GitHub

File: jv-oop-advanced/src/main/java/core/basesyntax/Figure.java (hunk 6 above)
Comment (DankevichMisha) [reply jv-oop-advanced/src/main/java/core/basesyntax/Figure.java:30]: searching mistake , forgot to delete

File: jv-oop-advanced/src/main/java/core/basesyntax/Form.java (hunk 9 above)
Comment (DankevichMisha) [reply jv-oop-advanced/src/main/java/core/basesyntax/Form.java:7]: okay

File: jv-oop-advanced/src/main/java/core/basesyntax/FigureSupplier.java (hunk 8 above)
Comment (DankevichMisha) [reply jv-oop-advanced/src/main/java/core/basesyntax/FigureSupplier.java:23]: +

File: jv-oop-advanced/src/main/java/core/basesyntax/IsoscelesTrapezoid.java (hunk 10 above)
Comment (DankevichMisha) [reply jv-oop-advanced/src/main/java/core/basesyntax/IsoscelesTrapezoid.java:4]: need to delete this lines?

File: jv-oop-advanced/src/main/java/core/basesyntax/Main.java (hunk 11 above)
Comment (DankevichMisha) [reply jv-oop-advanced/src/main/java/core/basesyntax/Main.java:5]: +

File: jv-oop-advanced/src/main/java/core/basesyntax/Main.java (hunk 12 above)
Comment (DankevichMisha) [reply jv-oop-advanced/src/main/java/core/basesyntax/Main.java:11]: like this?

for (int i = 0; i < 6; i++) { // констант
            if (i < 3) {
                figures[i] = FigureSupplier.getRandomFigure();
            }
            if (i >= 3) {
                figures[i] = FigureSupplier.getDefaultFigure();
            }
            figures[i].draw();
        }

File: jv-oop-advanced/target/checkstyle-cachefile (hunk 14 above)
Comment (DankevichMisha) [reply jv-oop-advanced/target/checkstyle-cachefile:1]: +

**Comment by DankevichMisha** [comment]
.

**Review by Rommelua — CHANGES_REQUESTED** [review]
To many files 
image

**Pull request closed without merging**